### PR TITLE
Spring shopping cart

### PR DESCRIPTION
**Pull request by keda-vlad**

### CODE REVIEW

**Review by ConfusedSatlan — APPROVED** [review]

File: src/main/java/mate/academy/bookstore/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addToCart(AddToCartRequestDto requestDto, Long userId) {
+        User user = userRepository.findById(userId).orElseThrow(() ->
+                new EntityNotFoundException("Can't find user by id = " + userId));
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id = " + requestDto.bookId()));
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId).orElseGet(() -> {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/bookstore/service/impl/ShoppingCartServiceImpl.java]: better extract it to a separate method (for example: ShoppingCart shoppingCart = getOrCreateShoppingCart())

File: src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+              - column:
+                  name: user_id
+                  type: BIGINT
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: BOOLEAN
+                  constraints:
+                    nullable: false
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/10-create-shopping-carts-table.yaml]: empty line

File: src/main/resources/db/changelog/changes/11-create-cart-item-table.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+              - column:
+                  name: quantity
+                  type: INTEGER
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
+                  type: BOOLEAN
+                  constraints:
+                    nullable: false
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/11-create-cart-item-table.yaml]: empty line

File: src/main/resources/db/changelog/changes/11-create-cart-item-table.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            tableName: cart_item
+            columns:
+              - column:
+                  name: id
+                  type: BIGINT
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+              - column:
+                  name: shopping_cart_id
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/11-create-cart-item-table.yaml:16]: is it possible to add FK?

**Review by vovasalyha — APPROVED** [review]
Approving, but please, take a look on my comments

**Pull request merged**